Cingulara/openrmf-api-compliance
# develop

- updating the version
- updated README
- updating tests to go against specific branches
- Adding more fine tuned GH pipeline tests
- #31 caching certain API calls
- Updated tests for CI/CD in GH Actions
- #28 XLSX for Compliance generation
- #27 CCI item information returned
- adding Jaeger Tracing framework
- updating to latest version for POAM
- filter on major control for report by host
- Updating to version 0.12 on version
- putting the latest version tag in here
- Adding Jaeger Tracing Information
- adding prometheus metrics export
- updated copyright notice
- updated API logging
- updating the API documentation
- adding the copyright notice
- Naming the NATS client; removing the DB connection
- adjust where we call the compliance date update
- document the NATS call we send out
- Update the system compliance generation date
- allow checklists to link to system records
- Update issue templates
- #20 adding NATS client reconnect and options
- code cleanup
- #15 fixing incorrect namespace
- making the ENV all caps
- rework to use compliance request/reply NATS svc
- fixing code for RBAC and API to API comms
- adding NATS for reading checklists and system list
- RBAC AuthN and AuthZ with keycloak for version 0.8
- adding more unit tests
- Adding Unit Tests
- Adding more model tests
- Removing older STIG type code and adding tests
- Reorg for tests
- fixing the compliance listing
- Adding in PII filters
- Compliance for low/moderate/high
- updating to build latest only
- update for openRMF
- fixing rule compliance for overall status
- return the list of strings
- adding a CONTROL to CCI listing API
- fixing the data for CCI List
- updating for making dockerhub quicker
- updating the version to 0.6
- grouping compliance by family
- updating control matching to NIST
- Fixing errors in NIST controls
- fixing finding parent controls
- finding titles for controls not in NIST
- #1 adjusting titles for those not in NIST
- adjusting finding nist titles
- fixing compile problem :(
- adding in the title search on NIST controls
- updating the compliance listing
- #1 Fixing the sort order of items
- #1 fixing compliance listing
- #1 fixing the host name information
- #1 adding host info to the compliance
- fixing the version
- fixing the port on the makefile
- finalizing the listing of controls and order
- fixes to the compliance to run through ALL NIST controls
- Finalizing the ingest of the CCI to NIST XML
- updating the model to read in the CCI to NIST listing correctly
- updating the readme. Generatign the CCI listing class for NIST XML data
- Adding the initial cycling through of Vulns
- Adding the initial setup of getting compliance from each checklist
